File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        if (n == 0) return;

        int write = 0;
        for (int read = 0; read < n; read++) {
            if (nums[read] != 0) {
                nums[write] = nums[read];
                write++;
            }
        }
        for (int i = write; i < n; i++) {
            nums[i] = 0;
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = 0

        for i in range(len(nums)):
            if nums[i] != 0:
                nums[left], nums[i] = nums[i], nums[left]
                left += 1

'''
case: nums = [0]
'''

'''
case: nums = [0, 1, 2, 0]
left = 0
i = 0
i = 1
    nums = [1, 0, 2, 0]
    left = 1
i = 2
    nums = [1, 2, 0, 0]
    left = 2
'''